Apromore-Extras/Build-Tools/src/main/resources/checkstyle.xml

```diff
@@ -1,256 +1,223 @@
-        <module name="TodoComment">
-            <property name="format" value="INFO"/>
-        </module>
-        <module name="UpperEll"/>
-
-    </module>
-
-</module>
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+          "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<!--
+
+  Checkstyle configuration that checks the sun coding conventions from:
+
+    - the Java Language Specification at
+      http://java.sun.com/docs/books/jls/second_edition/html/index.html
+
+    - the Sun Code Conventions at http://java.sun.com/docs/codeconv/
+
+    - the Javadoc guidelines at
+      http://java.sun.com/j2se/javadoc/writingdoccomments/index.html
+
+    - the JDK Api documentation http://java.sun.com/j2se/docs/api/index.html
+
+    - some best practices
+
+  Checkstyle is very configurable. Be sure to read the documentation at
+  http://checkstyle.sf.net (or in your downloaded distribution).
+
+  Most Checks are configurable, be sure to consult the documentation.
+
+  To completely disable a check, just comment it out or delete it from the file.
+
+  Finally, it is worth reading the documentation.
+
+-->
+
+<!--
+
+  Comparison to the draft Apromore coding style at
+  https://apromore.atlassian.net/wiki/spaces/AP/pages/450691073/Apromore+coding+standards
+
+  1. Code formatting
+
+    a. The length of code lines should be within one page width
+       -> LineLength
+
+    b. Use indentation for code layout (replace tabs with 4 4 spaces)
+       -> FileTabCharacter
+
+  3. Functions/methods
+
+    a. Function should be small and cohesive. A one-page long function is not small.
+       -> MethodLength
+
+    d. Minimize the number of function arguments
+       -> ParameterNumber
+
+  5. Comments
+
+    c. Each file should have at least a license header and comment explaining the purpose, usage and/or relationship with other classes (look at the JDK)
+       -> Header (but we'll be using maven-license-plugin instead)
+       -> MissingJavadocType (see https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocType)
+
+-->
+
+<module name="Checker">
+    <!--
+        If you set the basedir property below, then all reported file
+        names will be relative to the specified directory. See
+        https://checkstyle.org/5.x/config.html#Checker
+
+        <property name="basedir" value="${basedir}"/>
+    -->
+
+    <property name="fileExtensions" value="java, properties, xml"/>
+
+    <!-- Checks that a package-info.java file exists for each package.     -->
+    <!-- See http://checkstyle.sf.net/config_javadoc.html#JavadocPackage -->
+    <!-- <module name="JavadocPackage"/> -->
+
+    <!-- Checks whether files end with a new line.                        -->
+    <!-- See http://checkstyle.sf.net/config_misc.html#NewlineAtEndOfFile -->
+    <module name="NewlineAtEndOfFile"/>
+
+    <!-- Checks that property files contain the same keys.         -->
```

Comment:
These conventions are much more than the coding standards. That's ok but suggest to add these conventions to the current coding standards. At least, a link to the checkstyle documentation site in the current standard would help anybody to have a good understanding than reading this checkstyle.xml. Btw, "http://checkstyle.sf.net" is obsolete, should be "https://checkstyle.org" to match with the maven checkstyle plugin site.

Reply:
Sometimes the checkstyle plugin passed without detecting any deliberate violations. It's rare, but did happen, keep watching. Anyway, this is a good tool.